src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -263,6 +265,39 @@ function mpas_geotile_mgr_init(mgr, path) result(ierr)
             endif
         endif
 
+        !
+        ! Some datasets describe their z dimension as either tile_z or tile_z_start
+        ! and tile_z_end. mpas_parse_index will return either one or the other. However,
+        ! we will need a tile_z value to pass to read_geogrid and we should allocated
+        ! each z coordinate of a tile in Fortran to be between tile_z_start and tile_z_end.
+        !
+        ! Currently, no static dataset that MPAS uses describes its z coordinate with a lowerbound
+        ! other than 1.
+        !
+        tile_nz => null()
+        tile_z_start => null()
+        tile_z_end => null()
+
+        call mpas_pool_get_config(mgr % pool, 'tile_z', tile_nz)
+        call mpas_pool_get_config(mgr % pool, 'tile_z_start', tile_z_start)
+        call mpas_pool_get_config(mgr % pool, 'tile_z_end', tile_z_end)
+
+        if (associated(tile_nz)) then
+            ! Here we are assuming that if tile_z is specified then tile_z_start and tile_z_end
+            ! are not. This is safe currently as no dataset that MPAS uses specifies both.
+            allocate(tile_z_start)
+            allocate(tile_z_end)
```

Comment:
Do these allocations result in a memory leak? If so, should we address those in this PR?

Reply:
It does. I have a branch, [init_atmsophere/geotile_memleak_fix](https://github.com/MiCurry/MPAS-Model/tree/init_atmosphere/geotile_memleak_fix), which has two commits that fix this issue and other memleaks in `mpas_geotile_manager` and in `mpas_parse_geoindex`. 

Previously I was thinking about making a PR for that branch after this one was made, but perhaps it would be easiest to pull those commits onto this one.

Reply:
If the `init_atmosphere/geotile_memleak_fix` branch fixes memory leaks besides this one, then I think a separate PR make sense. Otherwise, it might be cleanest to not introduce memory leaks here in the first place.

Reply:
It does. Sounds good. I'll push an update that fixes the `MPAS_LOG_CRIT` comment and removes these memory leaks.